### PR TITLE
Fix findLatestCheckPoint

### PR DESCRIPTION
**Pull request by yrong**
For the issue Alistair found with the execution relayer in rococo environment bricked.

```
May 01 22:29:34 ip-172-31-41-212 start-execution-relayer.sh[1475828]: {"@timestamp":"2024-05-01T22:29:34.93758737Z","blockRoot":"0xd272b2946a1a06f641f50fd96bfe316c29511ffaf0f9b5edd0c0dc3a587c14c6","computedRoot":"0xd272b2946a1a06f641f50fd96bfe316c29511ffaf0f9b5edd0c0dc3a587c14c6","level":"info","message":"beacon header with block included found","slot":4868811}
May 01 22:29:34 ip-172-31-41-212 start-execution-relayer.sh[1475828]: {"@timestamp":"2024-05-01T22:29:34.937627911Z","calculatedCheckpointSlot":0,"level":"info","message":"checkpoint slot not available, try with slot in next sync period instead"}
May 01 22:29:34 ip-172-31-41-212 start-execution-relayer.sh[1475828]: {"@timestamp":"2024-05-01T22:29:34.982432636Z","calculatedCheckpointSlot":4874240,"lastFinalizedSlot":4905664,"level":"info","message":"fetch checkpoint on chain backward from history"}
May 01 22:29:34 ip-172-31-41-212 start-execution-relayer.sh[1475828]: {"@timestamp":"2024-05-01T22:29:34.992440215Z","error":"fetch execution header proof: get header update by slot with ancestry proof: populate closest checkpoint: get history finalized header for the checkpoint: Can't find checkpoint on chain for slot 4868811","level":"fatal","message":"Unhandled error"}
```

### CODE REVIEW

File: relayer/relays/beacon/header/header.go [hunk 1]
```diff
@@ -436,11 +436,9 @@ func (h *Header) findLatestCheckPoint(slot uint64) (state.FinalizedHeader, error
...
 	startIndex := uint64(lastIndex)
 	endIndex := uint64(0)
-	if uint64(lastIndex) > h.protocol.Settings.EpochsPerSyncCommitteePeriod {
-		endIndex = endIndex - h.protocol.Settings.EpochsPerSyncCommitteePeriod
-	}
 
 	syncCommitteePeriod := h.protocol.Settings.SlotsInEpoch * h.protocol.Settings.EpochsPerSyncCommitteePeriod
+	slotPeriodIndex := slot / syncCommitteePeriod
 
 	for index := startIndex; index >= endIndex; index-- {
```
Comment (claravanstaden) [inline relayer/relays/beacon/header/header.go:443]: `endIndex` will also be `0`, is that correct?

**Review by claravanstaden — COMMENTED** [review]
Thanks for the fix, Ron! Could you please describe what the bug is here and your fix changes that?

File: relayer/relays/beacon/header/header.go (hunk 1 above)
Comment (yrong) [reply relayer/relays/beacon/header/header.go:443]: Yeah, that's primary change. _**By extending search range until the very beginning.**_

Basically what happened on Rococo setup is that we upgraded the beacon relayer a few weeks back, but we never upgraded the execution relayer so the bridge was down for about ~2-3 weeks.

So we need to search from the latest checkpoint backward until we find the matched checkpoint, previously we search for only 256 epochs at most which is not correct.

**Review by claravanstaden — APPROVED** [review]

**Review by alistair-singh — APPROVED** [review]
+1

**Pull request merged**